lib/Models/Definition/upsertModelFromJson.ts

```diff
@@ -78,11 +78,8 @@ export default function upsertModelFromJson(
 
   if (model === undefined) {
     try {
-      model = factory.create(
-        isJsonString(json.type) ? json.type : undefined,
-        uniqueId,
-        terria
-      );
+      const type = isJsonString(json.type) ? json.type : undefined;
+      model = factory.create(type, uniqueId, terria);
```

Comment:
I more meant that you know sending `type === undefined` to `factory.create` gives undefined, so why not:
```js
model = isJsonString(json.type) ? factory.create(json.type, uniqueId, terria) : undefined;
```

Reply:
Oh I see. Yeah that is much better

Reply:
Done!
Sorry I'm a bit slow today